dvc/repo/metrics/show.py

```diff
@@ -129,6 +129,8 @@ def _read_metrics(self, metrics, branch):
     res = {}
     for out, typ, xpath in metrics:
         assert out.scheme == "local"
+        if not typ:
+            typ = os.path.splitext(out.path.lower())[1].replace(".", "")
```

Comment:
why `.replace(".", "")` though?

Reply:
It is because to convert the `.extension` string to `extension`, e.g., `.json` to `json`.

Reply:
Ah, right, missed it :) Thanks!